feat(ErdosProblems): 884

kesslermaximilian opened the pull request:
Formalizes https://www.erdosproblems.com/884.
Closes #1001.

A few questions: I wasn't able to find mathlib functionality for the *ordered* list of divisors of a natural number, so I quickly wrote my own abbreviation.

Also, would it fit the project scope to also formalize the statement of Tao's paper, disproving this conjecture under assumption of hardy-littlewood? If so, there is currently only the general hardy-littlewood statement formalized in this repository, but Tao only requires a qualitative statement about existence, not density of admissable prime-tuples.

kesslermaximilian committed:
Add Erdos Problem 884

Formalizes https://www.erdosproblems.com/884.
Closes #1001.

kesslermaximilian committed:
Merge branch 'main' into erdos-884

Paul-Lez commented on FormalConjectures/ErdosProblems/884.lean:
```diff
@@ -0,0 +1,65 @@
+By convention, we set `divisors_increasing 0 = ∅`.
+As a `Finset`, this is the same as `Nat.divisors`
+-/
+abbrev divisors_increasing (n : ℕ) : List ℕ := (List.range (n + 1)).filter (· ∣ n)
```
I agree with Calle on this one. Note that you could also define this as `n.divisors.sort`.

kesslermaximilian committed:
use Nat.nth (· ∣ n) to pick out divisors of n

kesslermaximilian committed:
add implication of hardy-littlewood => erdos 884 false

kesslermaximilian committed:
Merge branch 'main' into erdos-884

kesslermaximilian committed:
Merge branch 'erdos-884' of github.com:kesslermaximilian/formal-conjectures into erdos-884

smmercuri commented on FormalConjectures/ErdosProblems/884.lean:
```diff
@@ -0,0 +1,86 @@
+Statement of Erdos conjecture 884. See `erdos_884` for the problem asking whether this is true.
+-/
+def erdos_884_stmt : Prop :=
+    sum_inv_of_divisor_pair_differences =O[Filter.atTop] (1 + sum_inv_of_divisor_pair_differences)
```
Shouldn't the RHS be `1 + sum_inv_of_consecutive_divisors`?

kesslermaximilian replied:
Oh yes, of course.
Thanks for catching that

kesslermaximilian committed:
fix: use correct sum on RHS

callesonne reviewed (commented):
Some more comments :)

kesslermaximilian committed:
Respond to PR review

- Write doc comment in LaTeX.
- Remove divisors_increasing definition
- Use notation for BigO
- Copy doc comment to problem statement as well

Paul-Lez commented on FormalConjectures/ErdosProblems/884.lean:
```diff
@@ -0,0 +1,92 @@
+see `erdos_884_false_of_hardy_littlewood` for this implication.
+However, the conjecture itself remains open.
+-/
+def erdos_884_stmt : Prop :=
```
(to stick a little closer to this repo's convention!)
```suggestion
def Erdos884 : Prop :=
```

kesslermaximilian replied:
It does so, indeed. The thing is just that I want to separate the statement as just a definition, and then have both the conjecture and the conditional disproof by Tao as two statements.

Paul-Lez reviewed (commented):
A few minor nitpicks and then this should be ready to merge!

kesslermaximilian committed:
respond to code review

mo271 reviewed (approved):
LGTM, thanks! 
One more suggestions (since we changed the order of `answer(sorry)` in the meantime.

Please click "resolve converstation" for those comments that are addressed. This makes review much easier.

kesslermaximilian commented on FormalConjectures/ErdosProblems/884.lean:
```diff
@@ -0,0 +1,93 @@
+-/
+@[category research open, AMS 11]
+theorem erdos_884 :
+    Erdos884 ↔ answer(sorry) := by
```
Makes sense!

kesslermaximilian committed:
Change order of answer(sorry)

Co-authored-by: Moritz Firsching <firsching@google.com>

kesslermaximilian committed:
Merge branch 'main' into erdos-884

kesslermaximilian committed:
update type (ℕ → ℚ) to (ℕ → ℝ)

The abbreviation for asymptotic limits was changed in main to only apply
to functions of type ℕ → ℝ anymore, so we need to adapt this as well.

kesslermaximilian commented:
No problem. I just noticed that in main, the notation for asymptotics got changed to require functions (ℕ → ℝ), so I updated this PR accordingly.
If the build succeeds (it does locally for me), then we can merge, I suppose @Paul-Lez ?

kesslermaximilian committed:
Merge branch 'main' into erdos-884

smmercuri reviewed (approved):
Sorry for the delay in review! This looks good to me outside some minor style comments

Paul-Lez committed:
Address review comments on Erdos 884
